include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -422,6 +422,34 @@ ORT_ALL_ARGS_NONNULL;
 ORT_API(const char*, OrtGetErrorMessage, _In_ const OrtStatus* status)
 ORT_ALL_ARGS_NONNULL;
 
+/**
+   * APIs to support non-tensor types - map and sequence.
+   */
+
+/**
+   * If input OrtValue represents a map, you need to retrieve the keys and values
+   * separately. Use index=0 to retrieve keys and index=1 to retrieve values.
+   * If input OrtValue represents a sequence, use index to retrieve the index'th element
+   * of the sequence.
+   */
+ORT_API_STATUS(OrtGetValue, const OrtValue* value, int index, OrtAllocator* allocator, OrtValue** out);
+
+/**
+   * Returns 2 for type map and N for sequence where N is the number of elements
+   * in the sequence.
+   */
+ORT_API_STATUS(OrtGetNumValues, const OrtValue* value, size_t* out);
```

Comment:
How about 'OrtGetValueCount'? We're not totally consistent in the C API (with 'OrtGetNumOfDimensions' instead of 'OrtGetDimensionCount') but I think more end in Count that return a count.

Reply:
Sure, will change it to count.